### DISCUSSION TITLE
two ideas for moving nodes by mouse (drag and drop)

### DISCUSSION THREAD

**Discussion by aaa1386** [freeplane/freeplane#1968] [Ideas]
When the mouse pointer is in drag and drop mode
FP does not use the folding mark. The function of FP is as follows: when the indicator activates the end part of the node, if we pause or if we drop without pausing, two different functions will be created. The user must always be careful which of the modes happens because the delay time is short. I am very careful, but many things go wrong.
My idea is very easy and logical, compare. It has a very clear point: the folding mark, when the node is folded, if it is clicked, it should unfold the node. We use the same rule. Look at the description inside the image.

Note: I have come up with a new idea that you can see in the next comments that I like the second idea more.The second idea is a very simple and logical idea

![image](https://github.com/user-attachments/assets/f2395ac3-2a8c-4a30-8cbc-f66072b61740)

**Comment by aaa1386** [comment]
There is also a problem in the function of now FP:
We have enabled the "Unfold node on paste" option, which means you want the folded nodes to remain: suppose node 1 is folded.
If you want this node to remain folded, you must drop node 2 to node 1 without delay.
This makes you unable to place the node to your desired node in the children of node 1.

**Comment by aaa1386** [comment]
Now the transfer of nodes is done with a different function that has its own advantages.
This is a different idea, I don't know if users like it or not?

**Comment by euu2021** [comment]
I like the idea of having the nodes automatically fold back after the dragged node is dropped. 

The main problem that it solves is that: 
- during the drag and drop operation, I do errand movements until I decide in which target node I will drop the dragged node
- so, a lot of nodes unfold during those indecisive errand movements that I make, which results in a mess of many nodes unfolded
- so, to restore the same folding state that I had before the drag and drop operation, I have to manually fold a couple of nodes

A mixed solution would be to fold all the nodes that are not the actual target node where the drop happened; but, keep that target node unfolded. But, I prefer the solution "revert back to the exact same folding state that was active when the drag and drop operation started".

**Comment by quickfold** [comment]
I agree that this function would be useful.

**Reply by aaa1386** [reply]
> I agree that this function would be useful.

I came up with a newer idea that I like this second idea more, what do you think?

![image](https://github.com/user-attachments/assets/c4eae44a-c3a3-43a4-86d2-6f2518e3d1e9)

**Reply by aaa1386** [reply]
This idea is very simple and logical. The previous idea was a bit complicated.

**Reply by aaa1386** [reply]
We can even simplify the idea.
Remove the drag and drop function on Foldink Mark
